application/common/components/adapters/SagePeopleIdStore.php

```diff
@@ -179,9 +179,17 @@ private function getRecordsFromResponseBody($body)
             \Yii::error($this->getIdStoreName() . ' responded with incomplete data set');
         }
 
-        return  array_map(
+        return array_map(
             function ($item) {
-                return DotNotation::collapse($item);
+                $properties = DotNotation::collapse($item);
+
+                /*
+                 * Drop any non-word characters and replace spaces with underscores.
+                 */
+                $username = str_replace(' ', '_', $properties[self::PROP_DISPLAY_NAME]);
```

Comment:
I think we'll want to do more sanitation than simply replacing spaces... it seems common (and thus somewhat expected) for a username to only contain letters/numbers/underscores (all of which are word characters). Are we using username for anything, and will other systems be receiving username from us? Then things like apostrophes and hyphens could potentially cause problems, depending on the expectations of those systems... though hopefully not.

I'm curious... why was the "remove non-word characters" change undone?

Reply:
Good question...I don't know to which other systems the username is passed, but within IdP I think there are no problems with accepting "non-word" characters. 

As for why I reduced the filtering, a couple reasons: the realization that our handling of username doesn't care about content, discovery that in the current data set nothing unusual is present (only '.' and '-'), and the complexity of passing arrays to `preg_replace` was distasteful to me.

Having said that, I'll put it back by your recommendation.